### ISSUE TITLE
Get/update fails for item IDs that contain spaces

### ISSUE THREAD

**Issue by cdlate** [elastic/elasticsearch-php#1278]
### Summary
I want to update or get a record having id containing a space using the PHP client library.
Expected: The record having the correct id is returned/updated
Actual: The record is not found

### Code snippet of problem

**Test dataset:**

Kibana dev tools for inserts:

POST test_index/doc/_bulk
{"index": {"_index": "test_index", "_id": "test test"}}
{"field1": "value1"}

This will return data properly:
GET /test_index/_doc/test%20test

```
$params = [
  'index' => 'test_index',
  //'id'    => 'lp-3MIYB7NAUVXHg5pvt', THIS is working perfectly(no spaces)
  'id'    => 'test test', // this is not working because urlencode does not respect the url encode standard.
  'type' => '_doc'
];

$response = $elasticClient->get($params);
```

I think the problem is caused by the usage of urlencode instead of rawurlencode in the file AbstractEndPoint, not sure about how many client's methods are affected, not sure if the latest versions are affected, but I highly suspect they are.

See: https://www.php.net/manual/en/function.urldecode.php
See: https://www.php.net/manual/en/function.rawurlencode.php

### System details
- Ubuntu
- 7.3
- 6.x//7.x
- 7.17

**Comment by ezimuel** [comment]
@cdlate you right, the `+` character is not encoded anymore in a space [since ES 7.4](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/breaking-changes-7.4.html#_changes_to_encoding_plus_signs_in_urls). You can force to use `+` as space in ES 7.x with the settings `es.rest.url_plus_as_space` to `true`. 
We need to fix this as you proposed using `rawurlencode()` instead of `urlencode()`. At the same time, we need to avoid BC break since we cannot change the URL encoding for existing document (containing space in ID) indexed with `elasticsearch-php`.
I'm thinking to use an env variable like `ES_URL_PLUS_AS_SPACE` that will be `false` as default. We will use `rawurlencode()` if the previous env will be false, otherwise we will continue to use `urlencode()`.
Any feedback? Thanks!

**Comment by cdlate** [comment]
Hey @ezimuel
It is a good plan! I knew I was missing something about the evolution of the library, thanks for giving me context about it.

Can I contribute?

**Comment by ezimuel** [comment]
I sent #1303 for fixing the issue. @cdlate can you check if this work for you? Thanks.

**Comment by ezimuel** [comment]
@cdlate I just merged #1303 in 7.17 branch. I'll release soon 7.17.2
